native/src/main/java/io/ballerina/stdlib/persist/Utils.java

```diff
@@ -113,6 +122,21 @@ public static BArray[] getMetadata(RecordType recordType) {
         return new BArray[]{fieldsArray, includeArray, typeDescriptionArray};
     }
 
+    public static BMap<BString, Object> getFieldTypes(RecordType recordType) {
+        MapType stringMapType = TypeCreator.createMapType(PredefinedTypes.TYPE_STRING);
+        //TODO: use PredefinedTypes.TYPE_TYPEDESC once NPE issue is resolved
```

Comment:
```suggestion
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)